### ISSUE TITLE
Host FLASHViewer

### ISSUE THREAD

**Issue by timosachsenberg** [OpenMS/OpenMS#7201]

**Comment by timosachsenberg** [comment]
what is the state here?

**Comment by JeeH-K** [comment]
> what is the state here?

Now I have time to work on this!

**Comment by JeeH-K** [comment]
@axelwalter I need your insight for one thing. 
Andrew and I tested the docker file, and it works fine locally. The problem is that FLASHViewer is currently only implemented for local usage with "workspace" as a local directory. I have two options:
1. set the different branch for online deployment with streamlit-template equivalent workspace setting (need a bit of time to do so)
2. just make the workspace directory deleted every night (maybe privacy or security issue? since all users can access the uploaded files)

**Comment by axelwalter** [comment]
Just for testing the deployment the second option as you mention it would be fine, maybe with a warning that all files will be public.

The second option could be better if you remember which files a user uploaded via session state. Then only make these files available. However, this info will be lost even on page reload. Or let the user define a passphrase mapping the uploading files, which can then be re-entered to retrieve the files again.

For the user it would probably be nice if all OpenMS apps work similar with the first option.

**Comment by timosachsenberg** [comment]
I agree with Axel that basing FLASHViewer on the streamlit-template etc. should be the way to go if somehow possible. E.g., FLASHViewer takes the template but add some heavy customization on top (maybe with some extra folders etc.). I think learning how this is possible could help other developments a lot...

**Comment by JeeH-K** [comment]
Thanks for the opinions. I'll work on option 1, but it may not finish within this sprint.. But we are slowly working on it!

**Comment by JeeH-K** [comment]
PR opened: https://github.com/OpenMS/streamlit-deployment/pull/19

**Comment by JeeH-K** [comment]
Still WIP:
- [ ] change data structures for efficiency in javascript: parsing mzML -> directly to json
- [ ] add documentation on the webpages
- [ ] FLASHQuant: update input files for the viewer
